### PR TITLE
Add ecosystem page (Components v7)

### PR DESCRIPTION
**Pull request by hasparus**
### Background

image
### Description

This PR adds the ecosystem page and updates `@theguild/components` to 7.4.0.

We've already had a PR and a deploy preview with this page, so this should be pretty straightforward to review.

### CODE REVIEW

File: packages/web/docs/src/components/product-updates.tsx [hunk 1]
```diff
@@ -38,21 +38,25 @@ export async function getChangelogs(): Promise<Changelog[]> {
   return productUpdatesFolder
     .slice(1) // cut `_meta.ts` which always comes first
     .map(item => {
-      if (!item.children) {
+      if (!('children' in item)) {
```
Comment (hasparus) [inline packages/web/docs/src/components/product-updates.tsx:41]: The `.mjs` import now infers better type for the page map, and some constituents of the union don't have `children`, so we need to use `in` operator to check for it.

File: packages/web/docs/src/components/ecosystem-page/page.tsx [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { GotAnIdeaSection } from '../../components/got-an-idea-section';
```
Comment (dimaMachina) [inline packages/web/docs/src/components/ecosystem-page/page.tsx:1]:
```suggestion
import { GotAnIdeaSection } from '../got-an-idea-section';
```

File: packages/web/docs/src/components/ecosystem-page/page.tsx [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import { Page as LandingPageContainer } from '../../components/page';
```
Comment (dimaMachina) [inline packages/web/docs/src/components/ecosystem-page/page.tsx:2]:
```suggestion
import { Page as LandingPageContainer } from '../page';
```

File: packages/web/docs/src/components/got-an-idea-section.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  );
+}
+
+declare global {
+  interface Window {
+    $crisp?: {
+      push: (args: any[]) => void;
+    };
+  }
+}
```
Comment (dimaMachina) [inline packages/web/docs/src/components/got-an-idea-section.tsx:136]: can you move it to `src/env.d.ts` ?

Comment (hasparus) [reply packages/web/docs/src/components/got-an-idea-section.tsx:136]: Good idea.

File: packages/web/docs/src/components/product-updates.tsx [hunk 5]
```diff
@@ -38,21 +38,25 @@ export async function getChangelogs(): Promise<Changelog[]> {
...
+          description: item.frontMatter.description || '',
           route: item.route!,
         };
       }
       // Folder
-      const indexPage = item.children.find(item => item.name === 'index');
+      const indexPage =
+        'children' in item && item.children
+          ? item.children.find(item => item.name === 'index')
+          : null;
```
Comment (dimaMachina) [inline packages/web/docs/src/components/product-updates.tsx:58]:
```suggestion
      const indexPage =
        'children' in item && item.children && item.children.find(item => item.name === 'index');
```

File: packages/web/docs/src/pages/ecosystem.mdx [hunk 6]
```diff
@@ -0,0 +1,6 @@
+---
+title: The Ecosystem
+description: Everything you need to scale your API infrastructure
+---
+
+export { default } from '../components/ecosystem-page/page.tsx'
```
Comment (dimaMachina) [inline packages/web/docs/src/pages/ecosystem.mdx:6]:
```suggestion
export { default } from '../components/ecosystem-page/page'
```

File: packages/web/docs/src/components/got-an-idea-section.tsx (hunk 4 above)
Comment (hasparus) [reply packages/web/docs/src/components/got-an-idea-section.tsx:136]: BTW I created a component for this, but it's still in v8.

File: packages/web/docs/src/components/ecosystem-page/components.tsx [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    if (!product) {
+      throw new Error(`Product ${productName} is missing`);
+    }
+
+    return <ProductCard as="li" product={product} className="h-[222px]" {...props} />;
+  },
+};
+
+const ecosystemPageNav = {
+  ...Object.fromEntries(Object.keys(components).map(key => [key, () => null])),
```
Comment (dimaMachina) [inline packages/web/docs/src/components/ecosystem-page/components.tsx:49]:
```suggestion
  // ...
  ...Object.fromEntries(Object.keys(components).map(key => [key, () => null])),
```
just add comment why you hide `components`

**Review by dimaMachina — APPROVED** [review]

File: packages/web/docs/next.config.js [hunk 8]
```diff
@@ -10,6 +10,7 @@ export default withGuildDocs({
   eslint: {
     ignoreDuringBuilds: true,
   },
+
```
Comment (hasparus) [inline packages/web/docs/next.config.js:13]: accidental

**Pull request merged**